src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,172 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(Node<T> prev, T value, Node<T> next) {
+            this.prev = prev;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        if (head == null) {
+            addBeginElement(value);
+        } else {
+            addEndElement(value);
```

Comment:
```suggestion
            addTail(value);
```
Use the same words everywhere